src/test/java/test/io/ticktok/client/tick/HttpTickerPolicyTest.java

```diff
@@ -0,0 +1,185 @@
+import static org.junit.jupiter.api.Assertions.assertTimeoutPreemptively;
+import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.fail;
+import static test.io.ticktok.client.tick.HttpTickerPolicyTest.TicktokServer.PORT;
+
+public class HttpTickerPolicyTest {
+
+    public static final int SECOND = 1000;
+
+    final TickListener tickListener = new TickListener();
+    TicktokServer server;
+
+    @Test
+    void ignoreOnConnectionError() throws Exception {
+        // No server
+        assertNoTicksWereConsumed();
+    }
+
+    private void assertNoTicksWereConsumed() throws InterruptedException {
+        AtomicInteger tickCount = new AtomicInteger();
+        register("no-ticks", tickCount::incrementAndGet);
+        assertNoTicks(tickCount);
+    }
+
+    private void assertNoTicks(AtomicInteger tickCount) throws InterruptedException {
+        sleep(SECOND);
+        assertThat("Ticks were consumed", tickCount.get(), is(0));
+    }
+
+    private void register(String id, TickConsumer consumer) {
+        tickListener.forChannel(httpChannel(id)).register(consumer);
+    }
+
+    private TickChannel httpChannel(String id) {
+        Map<String, String> details = new HashMap<>();
+        details.put("url", format("http://localhost:%s/%s/pop", PORT, id));
+        return TickChannel.builder().type("http").details(details).build();
+    }
+
+    @Test
+    void ignoreOnRequestError() throws Exception {
+        server = new TicktokServer().onPopFailWith(500);
+        assertNoTicksWereConsumed();
+        assertTrue(server.popped);
+    }
+
+    @Test
+    void invokeOnTick() throws Exception {
+        server = new TicktokServer().tickOn("tick-me");
+        PacedTickCounter tickCounter = new PacedTickCounter(2);
+        register("tick-me", tickCounter::countDown);
+        tickCounter.assertNoTicks();
+        assertTimeoutPreemptively(ofSeconds(2), tickCounter::await, "No ticks");
+    }
+
+    @Test
+    void ignoreWhenThereNoTicksToPop() throws Exception {
+        server = new TicktokServer();
+        assertNoTicksWereConsumed();
+    }
+
+    @Test
+    void disconnectAllClocks() throws Exception {
+        server = new TicktokServer();
+        AtomicInteger tickCount = new AtomicInteger();
+        register("c1", tickCount::incrementAndGet);
+        register("c2", tickCount::incrementAndGet);
+        tickListener.disconnect();
+        sleep(100);
+        server.tickOn("c1", "c2");
+        assertNoTicks(tickCount);
+    }
+
+    @Test
+    void allowToRegisterAfterDisconnect() throws Exception {
+        server = new TicktokServer().tickOn("c1");
+        tickListener.disconnect();
+        CountDownLatch tickConsumed = new CountDownLatch(1);
+        register("c1", tickConsumed::countDown);
+        assertTimeoutPreemptively(ofSeconds(1), (Executable) tickConsumed::await);
+    }
+
+    @Test
+    void replaceCallbackForAGivenClock() throws Exception {
+        server = new TicktokServer();
```

Comment:
this test really not clear & clean, pls break to method with meaning full names.

Reply:
I tried, if you have a suggestion it would be great